common/display/displayqueue.cpp

```diff
@@ -536,6 +536,8 @@ bool DisplayQueue::QueueUpdate(std::vector<HwcLayer*>& source_layers,
     force_media_composition = true;
     applied_video_effect_ = requested_video_effect_;
     requested_video_effect = requested_video_effect_;
+    validate_layers = true;
```

Comment:
This is really full validation. Hmm we should be doing some fundamental mistake if we don't reset media state immediately. Did you check what happens when video is paused and video effect is disabled, why didn't we call SetMediaEffectsState ?

Reply:
If we dont do full validation, GetCachedLayers will get back only one single plane with all layers GPU composited when video is paused and therefore we don't get a chance to update video plane.

Reply:
k in this case video effects anyway will not be present right ? If all layers are gpu composited to a single plane we would go via the 3D Backend. Also, if we had multiple planes being used and now we suddenly have all layers for one plane than we should have already gone through validation. GetCachedLayers doesn't reset layer + plane combinations in anyway.

Reply:
I suspect if we actually ignore the update because layer damage is zero ? Can you make sure that the layer damage is set to displayframe in overlaylayer in case effects state has changed.

Reply:
The use case is as following:
1) Pause video playback and wait for a moment
2) HWC run into idle state, a ValidateLayers is requested with force_gpu set to true because of idle_frame.
3) previous_plane_state_ now contains a single plane with all layers GPU composited
4) GetCachedLayers check previous_plane_state_ and get back the sole plane
5) Video effect setting changed, but GetCachedLayers doesn't see any video plane.

To ensure a video plane is present again, a full validation is needed.

Reply:
Ya, in this case it's expected that Video effects don't have any impact. What we should do here is, DisplayQueue knows it's in idle mode. If it's in this mode and we get a request to change the video effect state we should put a request for self refresh to SurfaceFlinger. This should make sure the impact is impact and any frame update get combined. Also,  once we prepare idle_frame and the next update should actually call Validate layers to make sure we take optimal path for showing the layers https://github.com/intel/IA-Hardware-Composer/blob/master/common/display/displayqueue.cpp#L584. First check if in this case we have a bug here if not request a refresh to SurfaceFlinger once effect state changes.